### ISSUE TITLE
Saving a graph with an accent in title causes hiding all saved graphs, shows "no saved graphs" in tree "My Graphs".

### ISSUE THREAD

**Issue by fidergo-stephane-gourichon** [graphite-project/graphite-web#1979]
Hi

# Summary

Logged in user cannot see any saved graphs in "My Graphs" or "User Graphs".  Short URLs do work, suggesting that something is still registered.

# Steps to reproduce

* On Ubuntu 16.04 AMD64,
* installed packages `graphite-carbon` and `graphite-web`. Version is 0.9.15-1 .
* `graphite-manage syncdb` to create database and user.
* run test instance:

```
su -s /bin/bash -c '/usr/bin/django-admin runserver --settings \
 graphite.settings 0.0.0.0:8080' _graphite
```

* insert data point in database `{ echo "foo 1 1" | nc -q 0 127.0.0.1 2003 ; echo "foo 10 10" | nc -q 0 127.0.0.1 2003 ; }`
* go to http://0.0.0.0:8080/
* log in as newly created user.
* expand tree to reveal data 'foo'
* give a title to the graph with an accent, like "Écritures"
* click on disk icon to save graph, enter name "foo", click "ok".
* see window "Save to My Graphs - Result" "Graph saved successfully"

## Expected behavior

* tree display on left shows newly created graph

## Observed behavior

* tree display on left flashes with a new entry but reverts immediately to "No saved graphs"
* refreshing the page still shows "No saved graphs"

## Additional information

* Using the "Short URL" feature does provide a URL like this: http://0.0.0.0:8080/S/B that renders the graph, showing that something was stored on the server.
* I see URLs logged in terminal where `django-admin runserver` runs, no error

**Comment by fidergo-stephane-gourichon** [comment]
Turns out giving a title to the graph with an accent appeared to cause the problem.
Manually deleting the entry in carbon graph database allowed to see all previously saved graphs.

**Comment by deniszh** [comment]
Ah, good to know, thanks!

**Comment by fidergo-stephane-gourichon** [comment]
To see where is the problematic entry:

`sqlite3 /var/lib/graphite/graphite.db`

`select * from account_mygraph ;`

To delete it:

`delete from account_mygraph where ID=1`

**Comment by fidergo-stephane-gourichon** [comment]
We now know the actual cause and an efficient workaround, yet the issue is still fully present.  Also, when given a title with an accent, another consequence is that the graph rendering feature is immediately broken, the graph image is replaced with a small broken icon in top left.

**Comment by deniszh** [comment]
Hello @fidergo-stephane-gourichon,
Thanks for reporting this, but dashboard functionality of graphite-web is not really on top priority right now. Most of the people are using or simple graphs or 3rd party dashboards like 'Grafana'. If you have a proper fix - we would like to include it to current code, but I suspect that proper solution will require a quite big effort - maybe we can fix it later when we'll migrate to python3 and will have proper  UTF-8 support.

**Comment by brutasse** [comment]
Just filed a pull request to fix the issue, #1981 

Python 2 has full unicode support, but it requires more attention than with Python 3. Django handles unicode in requests/responses and in database models, so if there is a unicode bug it's very likely that it's in the Graphite-web codebase­…

Unicode support isn't going to come for free with Python 3, it's rather the opposite: Python 3 support will come for free when we fix the unicode sandwich in Graphite-web :)
